llvm/utils/update_mc_test_checks.py

```diff
@@ -0,0 +1,349 @@
+"""
+A test update script.  This script is a utility to update LLVM 'llvm-mc' based test cases with new FileCheck patterns.
+"""
+
+from __future__ import print_function
+
+import argparse
+import os  # Used to advertise this file's name ("autogenerated_note").
+
+from UpdateTestChecks import common
+
+import subprocess
+import re
+
+mc_LIKE_TOOLS = [
+    "llvm-mc",
+    "not llvm-mc",
+]
+
+ERROR_RE = re.compile(r"(warning|error): .*")
+ERROR_CHECK_RE = re.compile(r"# COM: .*")
+OUTPUT_SKIPPED_RE = re.compile(r"(.text)")
+COMMENT = {"asm": "//", "dasm": "#"}
+
+
+def invoke_tool(exe, cmd_args, testline, verbose=False):
+    if isinstance(cmd_args, list):
+        args = [applySubstitutions(a, substitutions) for a in cmd_args]
+    else:
+        args = cmd_args
+
+    cmd = 'echo "' + testline + '" | ' + exe + " " + args
+    if verbose:
+        print("Command: ", cmd)
+    out = subprocess.check_output(cmd, shell=True)
+    # Fix line endings to unix CR style.
+    return out.decode().replace("\r\n", "\n")
+
+
+# create tests line-by-line, here we just filter out the check lines and comments
+# and treat all others as tests
+def isTestLine(input_line, mc_mode):
+    # Skip comment lines
+    if input_line.strip(" \t\r").startswith(COMMENT[mc_mode]):
+        return False
+    elif input_line.strip(" \t\r") == "":
+        return False
+    # skip any CHECK lines.
+    elif common.CHECK_RE.match(input_line):
+        return False
+    return True
+
+
+def hasErr(err):
+    if err is None or len(err) == 0:
+        return False
+    if ERROR_RE.search(err):
+        return True
+    return False
+
+
+def getErrString(err):
+    if err is None or len(err) == 0:
+        return ""
+
+    lines = err.split("\n")
+    # take the first match
+    for line in lines:
+        s = ERROR_RE.search(line)
+        if s:
+            return s.group(0)
+    return ""
+
+
+def getOutputString(out):
+    if out is None or len(out) == 0:
+        return ""
+    lines = out.split("\n")
+    output = ""
+
+    for line in lines:
+        if OUTPUT_SKIPPED_RE.search(line):
+            continue
+        if line.strip("\t ") == "":
+            continue
+        output += line.lstrip("\t ")
+    return output
+
+
+def should_add_line_to_output(input_line, prefix_set, mc_mode):
+    # special check line
+    if mc_mode == "dasm" and ERROR_CHECK_RE.search(input_line):
+        return False
+    else:
+        return common.should_add_line_to_output(
+            input_line, prefix_set, comment_marker=COMMENT[mc_mode]
+        )
+
+
+def getStdCheckLine(prefix, output, mc_mode):
+    lines = output.split("\n")
+    output = ""
+    for line in lines:
+        output += COMMENT[mc_mode] + " " + prefix + ": " + line + "\n"
+    return output
+
+
+def getErrCheckLine(prefix, output, mc_mode):
+    if mc_mode == "asm":
+        return (
+            COMMENT[mc_mode]
+            + " "
+            + prefix
+            + ": "
+            + ":[[@LINE-1]]:{{[0-9]+}}: "
```

Comment:
```suggestion
            + ":[[#@LINE-1]]:{{[0-9]+}}: "
```

Reply:
Would be good to use the captured column number here instead of {{[0-9]+}} as well